Doc/c-api/long.rst

```diff
@@ -368,36 +368,77 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
           // Success!
       }
       else {
-          // Overflow occurred, but 'bignum' contains a truncated value.
+          // Overflow occurred, but 'value' contains the truncated
+          // lowest bits of pylong.
       }
 
+   The above example may look *similar* to
+   :c:func:`PyLong_As* <PyLong_AsSize_t>`
+   but instead fills in a specific caller defined type and never raises an
+   error about of the :class:`int` *pylong*'s value regardless of *n_bytes*
+   or the returned byte count.
+
+   To get at the entire potentially big Python value, this can be used to
+   reserve enough space and copy it::
+
+      // Ask how much space we need.
+      Py_ssize_t expected = PyLong_AsNativeBits(pylong, NULL, 0, -1);
+      if (expected < 0) {
+          // A Python exception was set with the reason.
+          return NULL;
+      }
+      assert(expected != 0);  // Impossible per the API definition.
+      uint8_t *bignum = malloc(expected);
+      if (!bignum) {
+          PyErr_SetString(PyExc_MemoryError, "bignum malloc failed.");
+          return NULL;
+      }
+      // Safely get the entire value.
+      Py_ssize_t bytes = PyLong_AsNativeBits(pylong, bignum, expected, -1);
+      if (bytes < 0 || bytes > expected) {  // Be safe, should not be possible.
+          if (!PyErr_Occurred()) {
```

Comment:
```suggestion
          if (bytes < 0) {
```

One of the few things the C API WG agree on is that we should have consistent enough return values that `PyErr_Occurred` should never be necessary. This is true with this API, so I'd rather not encourage using it.

Your call whether it's better to duplicate the free/return lines in separate blocks.

Reply:
ironically i had it written that way in my client earlier but shortened it to this for brevity.  agreed, lets not give people the idea that PyErr_Occurred is ever a _good_ idea.